py/label_microservice/repo_specific_model_test.py

```diff
@@ -0,0 +1,54 @@
+"""Unittest for repo_specific_model. """
+import logging
+from unittest import mock
+import pytest
+
+from label_microservice import mlp
+from label_microservice import repo_specific_model
+from label_microservice import test_util
+
+def test_predict_labels():
+  """A unittest for predict labels.
+
+  This function mocks out the embedding service and the MLPWorker.
+  """
+  model = repo_specific_model.RepoSpecificLabelModel()
+
+  model._mlp_predictor = mock.MagicMock(spec=mlp.MLPWrapper)
+  model._mlp_predictor.predict_probabilities.return_value = [[.2, .9]]
+
+  model._label_names = ["label1", "label2"]
+  model._label_thresholds = [.5 , .5]
+  model._get_issue_embedding = mock.MagicMock()
+  model._get_issue_embedding.return_value = [(10, 10)]
```

Comment:
This test shouldn’t pass in the ideal case because the issue embedding is of the wrong shape?  The model that makes predictions on top of the issue embedding should expect an embedding of a fixed size, and this embedding is only 2D.   I’m surprised that this test passes?  Thoughts?

Reply:
This unittest is mocking out the MLPClassifier and the embedding service. So really all its testing is the logic in predict_issue_labels. i.e that we form a dictionary with label -> probability and then filter it based on the thresholds. Its not testing the actual inference code.